gui/halt_list_frame.cc

```diff
@@ -110,6 +113,49 @@ bool halt_list_frame_t::compare_halts(halthandle_t const halt1, halthandle_t con
 		case nach_typ: // sort by station type
 			order = halt1->get_station_type() - halt2->get_station_type();
 			break;
+		case nach_transfer: // sort by no of transfers in previous month
+		{
+			int id1 = halt1->get_finance_history( 1, HALT_ARRIVED) == 0 ||
+				halt1->get_finance_history( 1, HALT_DEPARTED) == 0 ? 0 : 1;
+			auto halt1_transfers = halt1->get_finance_history(id1, HALT_ARRIVED)
+				< halt1->get_finance_history(id1, HALT_DEPARTED) ?
+				halt1->get_finance_history(id1, HALT_ARRIVED) :
+				halt1->get_finance_history(id1, HALT_DEPARTED);
```

Comment:
I wonder whether this calculation really represents the number of transfer. The number of passengers whose final destination is this halt will be counted here.

Reply:
I mean the transfers are the amount of people that enter the halt and then leave the halt again right? So I'm assuming that it would be the smaller of those two values that represents the amount of people entering and leaving (thus transferring) at the halt.

One way to confirm this assumption is to compare these numbers to a summation of passengers getting off at a halt where the halt is its "via" halt.
However that can get a bit more difficult in normal routing since you'd need to check that "via" halt before the pax arrives at the halt, because once the pax arrives at the halt and it needs to "transfer" its "via" halt is updated.
It of course gets easier when using time routing since those pax already have their route more fixed and thus it should be possible to see if the pax is transferring while at the halt (correct me if I'm wrong on that implementation).

I guess a different way of implementing it, now that I think of it, you could add a new "finance_history" for the halt which counts each passenger that arrives at the halt and it is its "via" halt as a transfer 🤔
But the question then becomes if you'd count something like walking as transferring too...

Reply:
I thought "transfer" passengers is the passengers who used the halt just to change the train. But as you said, it is difficult to estimate the exact value of this.
I think `min(arrived_passengers, departed_passengers)` still works as an indicator of how much buzy the halt is. It may be enough just to sort with the departures or arrivals, since usually these two values should be roughly same.

Reply:
Yeah that is probably correct, altho I think that the minimum of the two solution as a measure of "throughput" or "business"-score is still the best option, since the departure/arrival values can be quite skewed depending on if the destinations associated with the halt give equal departure as arrival.
![image](https://github.com/user-attachments/assets/82d1dfb5-8bf5-495d-928d-b2b748e5ec7f)
For example for the stadium halt on one of my maps the arrivals are consistently arrivals than arrivals.
I'm not sure what is maybe better for a name in that case, but probably something like "throughput" would better tell what the haltlist is sorted by.

Reply:
"Throughput" sounds good to me as the name of this indicator. Please update the name of the properties and functions as well.